### DISCUSSION TITLE
Stop recording after given time or on trigger.

### DISCUSSION THREAD

**Discussion by theldoria** [namhyung/uftrace#1598] [Q&A]
I want to apply `uftrace` to a cyclically running application, which may never stop.
However, I want to be able to record anyway and get the results after:

- a given max. time and/or
- using a trigger (probably examined in a Lua or Pathon script), e.g. after a specific function was entered for the N'th time

I would really love if recording would be done into a cyclic buffer, so I could stop recording on an user defined trigger (a watchpoint of a variable examined at any function enter/leave would be enough for may use case).

My question: is something like that possible?

EDIT: spelling

**Comment by namhyung** [comment]
I think we have two options for now.

1. use signal trigger like `-S INT@finish` : Ctrl-C (to the target process) will finish the tracing
2. setup function trigger like `-T yourFunc@finish` : it will finish the tracing as yourFunc is called

**Comment by honggyukim** [comment]
Hi @theldoria, I also think the features you mentioned can be useful, but we need time to implement those features.

@namhyung From my experience, I feel like sending a signal is not user-friendly. I think time based trace on/off is more useful and easy to use as @theldoria suggested.

How about this usage? (when recording a trace for `t-fork`.)
```
$ uftrace record -T 100ms@trace_on -T 500ms@finish -- t-fork
```
We can distinguish if `<symbol>` is a function name or a time unit by checking its first character.

But it might take time for implementation because I don't have much time these days.

**Comment by honggyukim** [comment]
I've created an issue at #1699.

**Comment by alexmyczko** [comment]
there is the timeout command. timout seconds uftrace parameters...

**Reply by namhyung** [reply]
Right, then it can emulate the time-based trigger using a signal.  It seems `timeout` would send a TERM signal after the given time.

https://man7.org/linux/man-pages/man1/timeout.1.html
